### PR TITLE
Refactored phase-change-level tests

### PR DESCRIPTION
**Pull request by BelligerG**
Refactored the phase-change-level test, moving the code into a common function, then calling this function from the test_ functions with the relevant parameter.

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

CLA
 - [x] If a new developer, signed up to CLA

### CODE REVIEW

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/acceptance/test_phase_change_level.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5216898&bid=17104879)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- improver_tests/acceptance/test_phase_change_level.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5216898&bid=17104879)

File: improver_tests/acceptance/test_phase_change_level.py [hunk 1]
```diff
@@ -41,54 +41,27 @@
 run_cli = acc.run_cli(CLI)
 
 
-def test_snow_sleet(tmp_path):
-    """Test snow/sleet level calculation"""
+@pytest.mark.parametrize("phase_type", ("snow-sleet", "sleet-rain",
```
Comment (benfitzpatrick) [inline improver_tests/acceptance/test_phase_change_level.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5216898)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/acceptance/test_phase_change_level.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5216898&bid=17104879)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/acceptance/test_phase_change_level.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5216898&bid=17104879)

File: improver_tests/acceptance/test_phase_change_level.py [hunk 2]
```diff
@@ -41,54 +41,27 @@
...
-                    "wbti.nc",
-                    "orog.nc",
-                    "land_mask.nc")]
-    args = [*input_paths,
-            "--phase-change", "sleet-rain",
-            "--horizontal-interpolation", "False",
-            "--output", output_path]
+            "--phase-change"]
+    horiz_interp = "True"
+    if len(phase_type.split("-")) > 2:
```
Comment (neilCrosswaite) [inline improver_tests/acceptance/test_phase_change_level.py]: The problem here is what if we add another test.
Pytest.parameterize does take a list of tuples for the second argument
```
@pytest.mark.parametrize("phase_type", [("snow-sleet", False), ("sleet-rain", False), ("sleet-rain-unfilled", True)])
```

Comment (BelligerG) [reply improver_tests/acceptance/test_phase_change_level.py]: Good shout! I'll update it now

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/acceptance/test_phase_change_level.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5216898&bid=17104879)

File: improver_tests/acceptance/test_phase_change_level.py [hunk 3]
```diff
@@ -41,54 +41,27 @@
...
 
 
-def test_snow_sleet(tmp_path):
-    """Test snow/sleet level calculation"""
+@pytest.mark.parametrize("phase_type,horiz_interp",
+                         [("snow-sleet", "True"),
+                          ("sleet-rain", "True"),
+                          ("sleet-rain-unfilled", "False")])
+def test_phase_change(tmp_path, phase_type, horiz_interp):
+    """Test phase change level calculation"""
```
Comment (neilCrosswaite) [inline improver_tests/acceptance/test_phase_change_level.py]: This should probably be expanded to cover all three tests

File: improver_tests/acceptance/test_phase_change_level.py [hunk 4]
```diff
@@ -41,54 +41,27 @@
...
-                    "orog.nc",
-                    "land_mask.nc")]
-    args = [*input_paths,
-            "--phase-change", "sleet-rain",
-            "--horizontal-interpolation", "False",
-            "--output", output_path]
+            "--phase-change"]
+    if len(phase_type.split("-")) > 2:
+        phase_type = "-".join(phase_type.split("-")[:2])
+    args.extend([phase_type,
```
Comment (neilCrosswaite) [inline improver_tests/acceptance/test_phase_change_level.py]: Could this be rearranged  so that the  args all get declared at the same time?

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/acceptance/test_phase_change_level.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5216898&bid=17104879)

File: improver_tests/acceptance/test_phase_change_level.py [hunk 5]
```diff
@@ -41,54 +41,30 @@
...
-    """Test snow/sleet level calculation"""
+@pytest.mark.parametrize("phase_type,horiz_interp",
+                         [("snow-sleet", "True"),
+                          ("sleet-rain", "True"),
+                          ("sleet-rain-unfilled", "False")])
+def test_phase_change(tmp_path, phase_type, horiz_interp):
+    """Testing:
+        sleet/rain level
+        snow/sleet level
+        sleet/rain level leaving below orography points	unfilled.
```
Comment (LaurenceBeard) [inline improver_tests/acceptance/test_phase_change_level.py]:
```suggestion
        sleet/rain level leaving below orography points unfilled.
```
rouge tab

Comment (BelligerG) [reply improver_tests/acceptance/test_phase_change_level.py]: Sounds like a good idea, I'll sort it now

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver_tests/acceptance/test_phase_change_level.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5216898&bid=17104879)

**Pull request merged**